exportlegends.lua

```diff
@@ -1038,6 +1038,11 @@ dfhack.onStateChange[GLOBAL_KEY] = function(sc)
     if sc == SC_VIEWSCREEN_CHANGED and df.viewscreen_choose_game_typest:is_instance(dfhack.gui.getDFViewscreen(true)) then
         asyncexport.reset_state()
     end
+
+    -- Reset state when a world or map is loaded to ensure data remains current
```

Comment:
why map and not just world? isn't race specific to world?

Reply:
Changes can occur between loading the map and opening legends thanks to the `open-legends` tool. As fort mode could easily modify historical data, I thought it safest to ensure that path starts with a reset state.

Reply:
this change won't catch that specific case -- if a map is loaded, then the race of a histfig is changed (dangerous, btw!), then `open-legends` is run, this check won't catch it. I'm not certain that it's a case we *really* need to address, though. resetting on world load, though, is *very* important and should absolutely be done here.

Reply:
the only reason I'm being picky about this is that maps are loaded and unloaded all the time in adventure mode, and I don't think we should run more than we really have to

Reply:
I was originally worried about werebeasts changing the race of histfigs, but after some research that doesn't appear to occur. As we aren't particularly worried about the case of `open-legends`, removing the reset on map load seems safe to me.